apps/admin-gui/src/app/shared/components/dialogs/copy-sponsored-members-dialog/copy-sponsored-members-dialog.component.ts

```diff
@@ -0,0 +1,145 @@
+    @Inject(MAT_DIALOG_DATA) public data: CopySponsoredMembersDialogData,
+    private store: StoreService,
+    private membersService: MembersManagerService,
+    private notificator: NotificatorService,
+    private translate: PerunTranslateService,
+    private auth: GuiAuthResolver
+  ) {}
+
+  ngOnInit(): void {
+    this.attrNames = this.attrNames.concat(this.store.getLoginAttributeNames());
+    this.membersService
+      .getAllSponsoredMembersAndTheirSponsors(this.data.voId, this.attrNames)
+      .subscribe({
+        next: (members) => {
+          this.sponsoredMembers = members;
+        },
+      });
+    this.voSponsorsTarget = this.data.voSponsors;
+    this.voSponsorsSource = this.data.voSponsors;
+    this.isPerunAdmin = this.auth.isPerunAdmin();
+    // if not PERUNADMIN user cannot copy to anyone else than self, so do not allow selecting self as source sponsor
+    if (!this.isPerunAdmin) {
+      this.voSponsorsSource = this.voSponsorsSource.filter(
+        (voSponsor) => voSponsor.id !== this.store.getPerunPrincipal().user.id
+      );
+    }
+    this.theme = this.data.theme;
+  }
+
+  sourceSponsorChanged(user: User): void {
+    this.tableLoading = true;
+    this.sourceSponsor = user;
+    this.voSponsorsTarget = this.data.voSponsors.filter(
+      (sponsor) => sponsor.id !== this.sourceSponsor.id
+    );
+    // disable copying to self only if PERUNADMIN (it is not possible to choose self as source sponsor without PERUNADMIN anyway)
+    if (this.isPerunAdmin) {
+      this.disableSelf = this.store.getPerunPrincipal().user.id === this.sourceSponsor.id;
+    }
+    this.sourceSponsorSelected = true;
+    this.selection.clear();
+    this.filteredSponsoredMembers = this.sponsoredMembers.filter((member) =>
+      member.sponsors.map((sponsor) => sponsor.user.id).includes(this.sourceSponsor.id)
+    );
+    this.tableLoading = false;
+  }
+
+  sponsor(members: MemberWithSponsors[]): void {
+    const sponsor =
+      this.sponsorType === 'self' ? this.store.getPerunPrincipal().user : this.targetSponsor;
+
+    const memberIds = members.map((member) => member.member.id);
+
+    this.membersService
+      .copySponsoredMembers(
+        memberIds,
+        this.sourceSponsor.id,
+        sponsor.id,
+        !this.pickExpiration,
+        this.expiration
+      )
+      .subscribe({
+        next: () => {
+          this.notificator.showSuccess(
+            this.translate.instant('DIALOGS.COPY_SPONSORED_MEMBERS.SUCCESS')
+          );
+          this.loading = false;
+          this.dialogRef.close(true);
+        },
+        error: () => (this.loading = false),
+      });
+  }
+
+  setExpiration(newExpiration: string): void {
+    if (newExpiration === 'never') {
+      this.expiration = 'never';
+    } else {
+      this.expiration = formatDate(newExpiration, 'yyyy-MM-dd', 'en-GB');
+    }
+  }
+
+  submit(): void {
+    this.loading = true;
+    const members = Array.from(this.selection.selected);
```

Comment:
It there a reason to copy the selected members ?

Reply:
Nope